designs/spot-consolidation.md

```diff
@@ -0,0 +1,105 @@
+
+Conceptually, this approach is straightforward to explain to customers and directly aligned with Spot best practices. Some customers may be disappointed by the flexibility requirement, but given the current lack of support for spot consolidation, it’s net positive from the status quo. While there may be valid use cases for customers to trade interruption rate for lower prices, we should delay enabling customers to configure this value until the use cases are better understood. 
+
+* 👍👍👍 Prevents consolidation from regressing to a lowest price allocation strategy
+* 👍 Easy to explain and implements spot best practices
+* 👍 Flexibility threshold is tunable, as best practices evolve. Cloud providers could choose to set it to 1 to disable this behavior.
+* 👎 Limits potential flexibility from status quo for both instance launch and replacement. Today, Karpenter can send unbounded flexibility to PCO (e.g. 100s of instance types). 
+* 👎 Fails to take advantage of opportunities to migrate to healthier pools due to limited instance type flexibility.
+
+### 2. Price Improvement Factor
+
+Karpenter could skip consolidation’s if the replacement instance types were only marginally better in price. For example, if replacing an m5a.large with an m5.large would only save 5%, it may not be worth the interruption. This behavior is valuable both for spot and on-demand instances, given the fact that consolidation is inherently an interruption. PIF must also constrain provisioning decisions, as without this, an instance could be provisioned by PCO, and then immediately consolidated. 
+
+Given the cost of interrupting a workload, as well as the relatively short average lifetime of a workload, this value would need to be significant (e.g. 2x). Further analysis of the right default value function (constant, linear, polynomial) is needed, and given the difficulty of choosing this value, we should think carefully before enabling customers to configure it themselves.
+
+This approach is predicated on customers configuring instance family flexibility, and relatively similar spot prices within a given size. For a customer flexible to all instance families within c,m,r categories, PCO would have a broad range of instance types to choose from, and would be unlikely to be forced to choose an unhealthy pool. This assumption follows EC2 spot best practices, and is relied upon by customers who use attribute-based selection with price protection.
+
+For customers who rely on instance size flexibility within a single family, or lack sufficient flexibility, they may face increased interruption rates compared to options 1 and 2. Due to the natural price differences between different instance sizes within the same family, this approach would limit flexibility to one or two instances sizes, depending on current spot markets. However, having limited flexibility goes against spot best practices, and the customer should expect increased interruptions.
```

Comment:
This same problem exists for the first option, I think. Assuming reduced flexibility on instance types, this may actually be better than option 1 since it would at least allow you to consolidate past a given price threshold